docs/api-communication-spec.md

```diff
@@ -0,0 +1,686 @@
+# PCAP Release API Documentation
+
+This document contains the API specification for the PCAP release.
+
+The pcap-release automates the identification of network capture targets based on metadata from:
+- **Cloud Foundry Applications**, where network traffic is captured directly from the application container.  
+- **BOSH** Deployment VMs, where network traffic is captured on the VM.
+
+Pcap-release uses gRPC for communication between all components. This document highlights the intended use of fields, scenarios considered by the implementation and detailed explanation of the data structures.
+
+## CF Case Overview
+
+Capturing data from cloud foundry applications requires the pcap-agent inside the app container. The pcap-agent runs in plain-text HTTP/2 (H2C). The Envoy in the app container provides the TLS termination.
+
+The Cloud Controller is used to identify the target ports of the pcap-agent in the target app instance.
+
+```mermaid
+graph LR
+    pcap-cli -->|H/2, gRPC| gorouter
+    gorouter -->|H/2, gRPC| pcap-api
+
+    subgraph pcap deployment
+        pcap-api
+    end
+    
+    subgraph diego-cell1
+        subgraph cf-app1
+            pcap-api -->|H/2, gRPC| c1e[Envoy]
+            c1e -->|gRPC, H2C| c1[pcap-agent]
+        end 
+        subgraph cf-app2
+            pcap-api -->|H/2, gRPC| c2e[Envoy]
+            c2e -->|gRPC, H2C| c2[pcap-agent]
+        end 
+    end
+    subgraph diego-cell2
+        subgraph cf-app3
+            pcap-api -->|H/2, gRPC| c3e[Envoy]
+            c3e -->|H2C, gRPC| c3[pcap-agent]
+        end 
+    end
+    subgraph cloud-controller
+        pcap-api -->|HTTP/1.1, JSON| cc[Cloud Controller]
+    end
```

Comment:
I'm not 100% sure but I thought we also talk to the cf-uaa for token validation? Or do we assume implicit validation if the CC accepts it (which is probably good enough)?

Reply:
I'll check again in the prototype.

Reply:
`cf cli` will request a token once from UAA at login step. Then, every request sent to Cloud Controller will embed that token in headers. 
The Cloud Controller does have a UAA client, but it doesn’t contact UAA to validate tokens on each request.